src/nvim/testdir/test_lambda.vim

```diff
@@ -23,29 +23,27 @@ function! Test_lambda_with_timer()
     return
   endif
 
-  source load.vim
-
   let s:n = 0
   let s:timer_id = 0
   function! s:Foo()
-    "let n = 0
-    let s:timer_id = timer_start(50, {-> execute("let s:n += 1 | echo s:n", "")}, {"repeat": -1})
+    let s:timer_id = timer_start(10, {-> execute("let s:n += 1 | echo s:n", "")}, {"repeat": -1})
   endfunction
 
   call s:Foo()
-  sleep 210m
+
   " do not collect lambda
   call test_garbagecollect_now()
-  let m = LoadAdjust(s:n)
-  sleep 230m
-  call timer_stop(s:timer_id)
 
-  let n = LoadAdjust(s:n)
-  let nine = LoadAdjust(9)
-
-  call assert_true(m > 1)
-  call assert_true(n > m + 1)
-  call assert_true(n < nine)
+  let m = s:n
+  " Wait for next timer callback.
+  for i in range(0, 10)
+    sleep 10m
+    if s:n > m
+      break
+    endif
+  endfor
+  call timer_stop(s:timer_id)
```

Comment:
might need to sleep once more after this, in order for the test to be correct. otherwise LGTM

Reply:
I assume the test is for `test_garbagecollect_now` to not remove the lambda, and `s:n` being incremented afterwards ensures that.
But I can see that we should maybe test that it is invoked twice really.
Do you mean something else?